src/transaction/deserialize.c

```diff
@@ -1,62 +1,71 @@
-/*****************************************************************************
- *   Ledger App Boilerplate.
- *   (c) 2020 Ledger SAS.
- *
- *  Licensed under the Apache License, Version 2.0 (the "License");
- *  you may not use this file except in compliance with the License.
- *  You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- *  Unless required by applicable law or agreed to in writing, software
- *  distributed under the License is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *  See the License for the specific language governing permissions and
- *  limitations under the License.
- *****************************************************************************/
+#include <stdbool.h>  // bool
+#include <string.h>   // memset, explicit_bzero, memmove
 
-#include "deserialize.h"
-#include "utils.h"
-#include "types.h"
-#include "../common/buffer.h"
 
-parser_status_e transaction_deserialize(buffer_t *buf, transaction_t *tx) {
-    if (buf->size > MAX_TX_LEN) {
-        return WRONG_LENGTH_ERROR;
-    }
-
-    // nonce
-    if (!buffer_read_u64(buf, &tx->nonce, BE)) {
-        return NONCE_PARSING_ERROR;
-    }
+#include "deserialize.h"
 
-    tx->to = (uint8_t *) (buf->ptr + buf->offset);
+#include "os.h" // THROW
+// #include "cx.h"
+#include "../sw.h"
+// #include "../common/read.h"
+#include "../common/buffer.h"
 
-    // TO address
-    if (!buffer_seek_cur(buf, ADDRESS_LEN)) {
-        return TO_PARSING_ERROR;
+/**
+ * XXX: considering only P2PKH, without timelock
+ * Validates that a script has the format of P2PKH. Throws an exception if doesn't.
+ * P2PKH scripts have the format:
+ *   [OP_DUP, OP_HASH160, pubkey_hash_len, pubkey_hash, OP_EQUALVERIFY, OP_CHECKSIG]
```

Comment:
It would be nice to have the length of each part in this comment.

Reply:
Done.

Reply:
Great. Thanks!